doc/releases/changelog-dev.md

```diff
@@ -170,6 +170,9 @@ featuring a `simulate` function for simulating mixed states in analytic mode.
   observables that are also unsupported by the device.
   [(#6653)](https://github.com/PennyLaneAI/pennylane/pull/6653)
 
+* Improved performance of ``iterative_qpe`` tests. Removed the overwhelming shots with exact solution via `qml.probs`.
+  [(#6683)](https://github.com/PennyLaneAI/pennylane/pull/6683)
+
```

Comment:
Since it's strictly a test change, we don't technically need a changelog entry, but I'm fine either way.

Reply:
@KetpuntoG